.test-infra/jenkins/Infrastructure.groovy

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as common
+import CommonTestProperties.SDK
+
+class Infrastructure {
+
+  static void prepareSDKHarness(def context, SDK sdk, String repositoryRoot, String dockerTag) {
+    context.steps {
+      String sdkName = sdk.name().toLowerCase()
+      String image = "${repositoryRoot}/${sdkName}"
+      String imageTag = "${image}:${dockerTag}"
+
+      shell("echo \"Building SDK harness for ${sdkName} SDK.\"")
+      gradle {
+        rootBuildScriptDir(common.checkoutDir)
+        common.setGradleSwitches(delegate)
+        tasks(":beam-sdks-${sdkName}-container:docker")
+        switches("-Pdocker-repository-root=${repositoryRoot}")
+        switches("-Pdocker-tag=${dockerTag}")
+      }
+      shell("echo \" Tagging Harness' image\"...")
+      shell("docker tag ${image} ${imageTag}")
+      shell("echo \" Configure docker credentials\"...")
+      shell("docker-credential-gcr configure-docker")
+      shell("echo \" Pushing Harness' image\"...")
+      shell("docker push ${imageTag}")
+    }
+  }
+
+  static void prepareFlinkJobServer(def context, String repositoryRoot, String dockerTag) {
+    context.steps {
+      String image = "${repositoryRoot}/flink-job-server"
+      String imageTag = "${image}:${dockerTag}"
+
+      shell('echo "Building Flink job Server"')
+
+      gradle {
+        rootBuildScriptDir(common.checkoutDir)
+        common.setGradleSwitches(delegate)
+        tasks(":beam-runners-flink_2.11-job-server-container:docker")
+        switches("-Pdocker-repository-root=${repositoryRoot}")
+        switches("-Pdocker-tag=${dockerTag}")
+      }
+
+      shell("echo \" Tagging Flink Job Server's image\"...")
+      shell("docker tag ${image} ${imageTag}")
+      shell("echo \" Configure docker credentials\"...")
+      shell("docker-credential-gcr configure-docker")
```

Comment:
@yifanzou @tvalentyn since yesterday you seem to have some more background on this ( 😃): is this safe to use here (won't this affect other jobs and credentials configuration)?

Reply:
It should be fine to use this command. This will not affect other jobs base on our experiments.

The problem was that I've run the 'gcloud auth configure-docker' on the build workers while installing the GCR credential helper. It wrote an additional incompatible creds pair "marketplace.gcr.io": "gcloud" into the docker config. I reset the config files on build workers so that all affected tests were executed successfully.

Reply:
It's only required to do once. It's already setup on the workers, so you don't have to run this every time.

Reply:
+1 to @yifanzou's comments.

Reply:
Great, thanks for clarifying!